### ISSUE TITLE
Investigate Divi conflict

### ISSUE THREAD

**Issue by joshfeck** [eventespresso/event-espresso-core#814]
Please see HS thread:

https://secure.helpscout.net/conversation/727763885/596100?folderId=100570

**Comment by joshfeck** [comment]
A quick recap on some troubleshooting I did:

Ee 4.9.69 + divi 3.18.3 + wp 5.0: Divi's front end editor works
EE current master + divi 3.18.3 + wp 5.0: Divi's front end editor is broken, JS errors
EE current master + divi 3.18.3 + wp 4.9.8 (no Gutenberg plugin activated) Divi's front end editor works

**Comment by nerrad** [comment]
I spent a bit of time tonight looking into this.  The frustrating thing here is that without having access to the divi source, I can't really troubleshoot this effectively but here's what I've discovered:

- We extract webpack boilerplate into its own file that is then loaded as a dependency for our built files.  Since the boilerplate nearly never changes, this helps keep that cached long term in people's browsers and reduces (by about 1-2kb) the size of each of our bundles.  I discovered that divi throws up on this.  Could be because divi also builds their `bundle.js` file through webpack only its a different version of webpack maybe.  So if I switch our build process to not extract this built file divi doesn't blow up. However...
- Although divi loads, there's a whole bunch of errors in the console for things like `_` not being found (lodash symbol) and `wp` not being found (wp js global).  What's frustrating here is all these errors have divi's `bundle.js` as the source of the errors in the stack trace.  So on the surface it appears like divi is doing something in maybe reading the scripts on the page or mirroring behaviour or something but whatever its doing its clobbering other things loading on the page.  It's all conjecture though without seeing the source 😠 
- I was able to tweak our script so it JUST loads the base file and enqueues wp core's `lodash` as a dependency.  This confirms the conflict here because divi throws undefined `_` errors (which clearly does NOT come from our code, I ensured there was none of our own javascript loading).  This will be because WP core loads lodash in no conflict mode whenever its library is enqueued:

See `wp-includes/script-loader.php`
```
did_action( 'init' ) && $scripts->add_inline_script( 'lodash', 'window.lodash = _.noConflict();' );
```

So at a minimum, divi is not accounting for the fact that on the frontend the WordPress lodash library might be in use.

I suspect this might be a cause for some of the other console errors I was seeing as well.

## Next steps

* [ ] investigate namespacing the webpack boilerplate (runtime) built for our files OR (what I suspect we'll just have to do) is just have the webpack runtime in each file like normal.
* [ ] For the short-term, don't load `ee-core-blocks-frontend.js` on the frontend.  It's not needed right now for anything and can potentially confuse the issue if divi investigates this and asks why we're loading that file (which would be a red herring for them).

For the short term that will solve the issues with divi, **HOWEVER** long term it is NOT a viable solution as eventually we will create blocks that require react and various WP dependencies to load on the frontend.  Ultimately, this does appear to be a divi theme issue because at a minimum, they are not handling conflicts with things like lodash or classnames which are in use by core (and presumably other plugins eventually that might list those wp core files as a dependency).

Even if I was to get access to divi source (which still would be a good thing to get), I don't think its a good use of our time (currently) to be fixing Divi code.

@joshfeck what do you think of this plan?

**Comment by nerrad** [comment]
I'm also going to see if I can find another plugin that enqueues react and/or wp assets on the frontend to see if that also causes issues with Divi.

**Comment by nerrad** [comment]
K I've created a pull that fixes the issue (see #815)
